infra/modules/monitoring/main.tf

```diff
@@ -0,0 +1,124 @@
+# Create CloudWatch alarms for ECS metrics
+
+resource "aws_cloudwatch_metric_alarm" "cpu" {
+  count                     = var.cpu_threshold != "" ? 1 : 0
+  alarm_name                = "${var.ecs_name}-cpu-usage"
+  comparison_operator       = "GreaterThanOrEqualToThreshold"
+  evaluation_periods        = 2
+  metric_name               = "CPUUtilization"
+  namespace                 = "AWS/ECS"
+  period                    = 120
+  statistic                 = "Average"
+  threshold                 = var.cpu_threshold
+  alarm_description         = "This metric monitors ecs cluster cpu utilization"
+  alarm_actions             = [aws_sns_topic.this.arn]
+  ok_actions                = [aws_sns_topic.this.arn]
+  insufficient_data_actions = [aws_sns_topic.this.arn]
+
+  dimensions = {
```

Comment:
This part is repeated a lot. Would it make sense to make it a local variable or something? I found myself wondering if we want more dimensions by default like region (in case the user is good and runs multi-regionally), and it'd be annoying to update every single one.

Reply:
Those alerts are tied up to ECS cluster that was passed to the monitoring module. Dimensions section declares which object exactly we are planning to monitor and alert. If we want to use monitoring module for some other ECS cluster in any other region we could simply call again this module and pass other ECS cluster as input variable.